src/transformStateWithClones.js

```diff
@@ -7,7 +7,27 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let newState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        newState = { ...newState, ...action.extraData };
+        break;
+
+      case 'removeProperties':
+        for (const removeKey of action.keysToRemove) {
+          delete newState[removeKey];
+        }
+        break;
+      case 'clear':
+        newState = {};
+    }
```

Comment:
[CODE STYLE]: Consider adding a default case in your switch statement for error handling. This ensures that if an action type is not recognized, the function can handle it appropriately, perhaps by throwing an error or ignoring the action.